posthog/batch_exports/service.py

```diff
@@ -847,3 +849,46 @@ async def afetch_batch_export_runs_in_range(
     ).order_by("data_interval_start")
 
     return [run async for run in queryset]
+
+
+def fetch_earliest_backfill_start_at(
+    *,
+    team_id: int,
+    model: str,
+    interval_time_delta: dt.timedelta,
+    exclude_events: collections.abc.Iterable[str] | None = None,
+    include_events: collections.abc.Iterable[str] | None = None,
+) -> dt.datetime:
+    """Get the earliest start_at for a batch export backfill.
+
+    Note that this query will return the unix epoch if no events or persons exist.
```

Comment:
Seems like we are not handling this case?

Reply:
So it will work as is (i.e. it won't raise any errors) but it won't return an error to the user to say there is no data to backfill. So yes, you're probably right, it would make sense to return an error to the front end to say there is no data and hence a backfill is unnecessary